tracee-ebpf/tracee/tracee.go

```diff
@@ -906,6 +909,157 @@ func (t *Tracee) initBPF() error {
 	return nil
 }
 
+func (t *Tracee) getProcessCtx(hostTid int) (external.ProcessCtx, error) {
+	processCtx, procExist := processTreeMap[hostTid]
+	if procExist {
+		//fmt.Printf("network process data: pid: %d, ctime: %v, ppid: %v, cgroup_id: %v\n",data.Pid, time.Unix(0, int64(data.Ctime+t.bootTime)), data.Ppid, data.Cgroup_id)
+		return processCtx, nil
+	} else {
+
+		processContextMap, err := t.bpfModule.GetMap("process_context_map")
+		if err != nil {
+			t.Close()
+			return processCtx, err
+		}
+		processCtxBpfMap, err := processContextMap.GetValue(unsafe.Pointer(&hostTid))
+		if err != nil {
+			return processCtx, err
+		}
+		processCtx = external.SetProcessContext(processCtxBpfMap)
+		return processCtx, nil
+	}
+}
+
+func timespecToTime(ts syscall.Timespec) time.Time {
+	return time.Unix(int64(ts.Sec), int64(ts.Nsec))
+}
+func getFileCtime(path string) (time.Time, error) {
+	var ctime time.Time
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return ctime, err
+	}
+	stat_t := fileInfo.Sys().(*syscall.Stat_t)
+	return timespecToTime(stat_t.Ctim), nil
+}
+
+func parseProcStatus(status []string, taskName string) (external.ProcessCtx, error) {
+	var process external.ProcessCtx
+	processFileds := []string{"Tgid:", "Pid:", "PPid:", "Uid:", "NStgid:", "NSpid:", "NSpgid:"}
+	i := 0
+	var processVals []int
+	for idx, val := range status {
+		if val == processFileds[i] {
+
+			i++
+			filed, err := strconv.ParseUint(status[idx+1], 10, 32)
+			if err != nil {
+				return process, err
+			}
+			processVals = append(processVals, int(filed))
+		}
+		if i > len(processFileds)-1 {
+			break
+		}
+
+	}
+	processCtime, err := getFileCtime(taskName)
+	if err != nil {
+		return process, err
+	}
+	process.Ctime = int(processCtime.Unix())
+	process.HostPid = processVals[0]
+	process.HostTid = processVals[1]
+	process.HostPpid = processVals[2]
+	process.Uid = processVals[3]
+	process.Pid = processVals[4]
+	process.Tid = processVals[5]
+	process.Ppid = processVals[6]
```

Comment:
Instead of using numbers [0-6], it will be better to have names.
You can use a map instead of an array to save and get the values, which will be clearer